DSCResources/MSFT_xDhcpClient/MSFT_xDhcpClient.psm1

```diff
@@ -0,0 +1,173 @@
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData -StringData @'
+GettingDHCPClientMessage=Getting the DHCP Client on {1} interface "{0}".
+ApplyingDHCPClientMessage=Applying the DHCP Client on {1} interface "{0}".
+DHCPClientSetStateMessage=DHCP Client was set to the desired state {2} on {1} interface "{0}".
+CheckingDHCPClientMessage=Checking the DHCP Client on {1} interface "{0}".
+DHCPClientDoesNotMatchMessage=DHCP Client is not in the desired state {2} on {1} interface "{0}".
+InterfaceNotAvailableError=Interface "{0}" is not available. Please select a valid interface and try again.
+'@
+}
+
+function Get-TargetResource
+{
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String] $InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('IPv4', 'IPv6')]
+        [String] $AddressFamily,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Enabled', 'Disabled')]
+        [String] $State
+    )
+
+    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+        $($LocalizedData.GettingDHCPClientMessage) `
+        -f $InterfaceAlias,$AddressFamily `
+        ) -join '')
+
+    Test-ResourceProperty @PSBoundParameters
+    
+    $CurrentDHCPClient = Get-NetIPInterface `
+        -InterfaceAlias $InterfaceAlias `
+        -AddressFamily $AddressFamily
+
+    $returnValue = @{
+        State          = $CurrentDHCPClient.Dhcp
+        AddressFamily  = $AddressFamily
+        InterfaceAlias = $InterfaceAlias
+    }
+
+    $returnValue
+}
+
+function Set-TargetResource
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String] $InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('IPv4', 'IPv6')]
+        [String] $AddressFamily,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Enabled', 'Disabled')]
+        [String] $State
+    )
+
+    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+        $($LocalizedData.ApplyingDHCPClientMessage) `
+        -f $InterfaceAlias,$AddressFamily `
+        ) -join '')
+
+    Test-ResourceProperty @PSBoundParameters
+    
+    $CurrentDHCPClient = Get-NetIPInterface `
+        -InterfaceAlias $InterfaceAlias `
+        -AddressFamily $AddressFamily
+        
+    # The DHCP Client is in a different state - so change it.
+    if ($CurrentDHCPClient.DHCP -ne $State)
```

Comment:
The test will return false if the DHCP state is already in the same state that the resource wants it to be. So in this case we wouldn't want to run set anyway. E.g. only fire the set if the state is different.

This is really only a double check anyway - because Test-Resource should return $true if the state is already correct - and therefore set-targetresource wouldn't even fire. 

Does this make sense? I may be missing what you mean though (not enough coffee yet).